pygmt/base_plotting.py

```diff
@@ -984,3 +984,161 @@ def text(
 
                 arg_str = " ".join([fname, build_arg_string(kwargs)])
                 lib.call_module("text", arg_str)
+
+    @fmt_docstring
+    @use_alias(R="region", J="projection", B="frame", S="convention")
+    @kwargs_to_strings(R="sequence",)
+    def meca(
+        self,
+        spec,
+        scale,
+        lon=None,
+        lat=None,
+        depth=None,
+        convention=None,
+        plot_lon=None,  # FIXME: implement -C flag if plot location is specified
+        plot_lat=None,
+        text=None,
+        **kwargs,
+    ):
+        """
+        Plot focal mechanisms.
+
+        {aliases}
+
+        Parameters
+        ----------
+        spec: dict, 1D array, 2D array, or str
```

Comment:
We could be more specific about when this is a dictionary — e.g. "If `spec` is a dictionary, the following conventions are supported and need to be present as keys in the dictionary"